docs/dev/developer-guides/transactions/fee-model.md

```diff
@@ -39,3 +36,45 @@ So the tips to make the most out of the zkSync fee system are the following:
 - **Reuse as many storage slots as possible.** Only the state diff is published on Ethereum.
 - **Users should share as many storage slots as possible.** If 100 users update a storage slot of your contract in a single block, the diff will be published only once. In the future, we will introduce reimbursement for the users, so that the costs for updating shared storage slots are split between the users.
 - **Reuse the contract code if possible.** On Ethereum, avoiding constructor parameters and putting them into constants reduces some of the gas costs upon contract deployment. On zkSync the opposite is true: deploying the same bytecode for contracts, while changing only constructor parameters can lead to substantial fee savings.
+
+
+## Gas estimation during transaction: For custom and paymaster accounts
+
+On Ethereum, there is a constant of 21000 gas that should cover all the intrinsic costs of processing a transaction: checking the signature and updating the nonce of the account. 
+
+On zkSync Era, it's quite different, because we support custom accounts that may take a slightly different (typically a bit higher) number of gas than EOAs. 
+We provide a convenient way for anyone to estimate the cost of a transaction regardless of the type of account. 
+
+### Changes in the `validateTransaction`
+
+At present, the `validateTransaction` method fails if the signature is incorrect and returns `success=true` for a valid signature. The system entirely disregards the `returndata`. However, we migrated to a new approach, where the success of the `validateTransaction` method depends on it returning a magic string. Any other `returndata` will be deemed invalid and rejected by the system.
```

Comment:
> At present

It is not how it is done presently, this is how it used to be. The sentence says that "we migrated to a new approach" :)

Reply:
I rephrased to 

> The `validateTransaction` method fails if the signature is incorrect and returns `success=true` for a valid signature. The system entirely disregards the `returndata`. However, the success of the `validateTransaction` method depends on it returning a magic string. Any other `returndata` will be deemed invalid and rejected by the system.

@StanislavBreadless make sense?

Reply:
@krofax 

How it used to be:

- `validateTransaction` is considered successful whenever it does not revert (i.e. returns `success = true`). For invalid signatures the method reverts.

How it is now:

- `validateTransaction` is considered successful whenever it does not revert (i.e. returns `success = true`) AND it returns the magic string. For invalid signatures, the method does not revert, but it returns invalid magic.

A lot of time has passed since the change, so we can only describe the part of how it is now.

Also, saying that the returndata is disregarded, but then saying that it is used to check the magic sounds a bit contradictive, but up to you

Reply:
@StanislavBreadless I updated this part to be 

> The `validateTransaction` method is considered successful whenever it does not revert (i.e. returns `success = true`). And it returns the magic string. For invalid signatures, the method does not revert, but it returns invalid magic. However, the success of the `validateTransaction` method depends on it returning a magic string.

What do you think?